telemetry-common/src/main/java/com/sequenceiq/cloudbreak/telemetry/diagnostics/DiagnosticsOperationsService.java

```diff
@@ -0,0 +1,160 @@
+                orchestratorMetadata.getGatewayConfigs(), orchestratorMetadata.getNodes(), orchestratorMetadata.getExitCriteriaModel());
+        Set<String> unresponsiveHosts = allUnresponsiveNodes.stream()
+                .filter(filter::apply)
+                .map(Node::getHostname).collect(Collectors.toSet());
+        LOGGER.debug("Diagnostics collection salt validation operation has been started. parameters: '{}' stack id: {} ", parameters.toMap(), stackId);
+        if (CollectionUtils.isNotEmpty(unresponsiveHosts)) {
+            if (parameters.getSkipUnresponsiveHosts()) {
+                parameters.getExcludeHosts().addAll(unresponsiveHosts);
+                LOGGER.debug("Diagnostics parameters has been updated with excluded hosts. parameters: '{}', stack id: {}", parameters.toMap(), stackId);
+            } else {
+                throw new CloudbreakOrchestratorFailedException(
+                        String.format("Some of the hosts are unresponsive, check the states of salt-minions on the following nodes: %s",
+                                StringUtils.join(unresponsiveHosts, ',')));
+            }
+        }
+    }
+
+    public void vmDiagnosticsReport(String resourceCrn, DiagnosticParameters parameters) {
+        vmDiagnosticsReport(resourceCrn, parameters, null, null);
+    }
+
+    public void vmDiagnosticsReport(String resourceCrn, DiagnosticParameters parameters, UsageProto.CDPVMDiagnosticsFailureType.Value failureType,
+            Exception exception) {
+        if (parameters == null) {
+            LOGGER.debug("Skip sending diagnostics report as diagnostic parameter input is empty.");
+            return;
+        }
+        UsageProto.CDPVMDiagnosticsEvent.Builder eventBuilder = UsageProto.CDPVMDiagnosticsEvent.newBuilder();
+        if (exception != null) {
+            eventBuilder.setFailureMessage(StringUtils.left(exception.getMessage(), ERROR_MESSAGE_MAX_LENGTH));
+            eventBuilder.setResult(UsageProto.CDPVMDiagnosticsResult.Value.FAILED);
+        } else {
+            eventBuilder.setResult(UsageProto.CDPVMDiagnosticsResult.Value.SUCCESSFUL);
+        }
+        setIfNotNull(eventBuilder::setFailureType, failureType);
+        setIfNotNull(eventBuilder::setUuid, parameters.getUuid());
+        setIfNotNull(eventBuilder::setDescription, parameters.getDescription());
+        setIfNotNull(eventBuilder::setAccountId, parameters.getAccountId());
+        setIfNotNull(eventBuilder::setInputParameters, parameters.toMap().toString());
+        setIfNotNull(eventBuilder::setCaseNumber, parameters.getIssue());
+        setIfNotNull(eventBuilder::setResourceCrn, resourceCrn);
+        UsageProto.CDPVMDiagnosticsDestination.Value dest = convertUsageDestination(parameters.getDestination());
+        setIfNotNull(eventBuilder::setDestination, dest);
+        usageReporter.cdpVmDiagnosticsEvent(eventBuilder.build());
+    }
```

Comment:
something like:
```
    public void vmDiagnosticsReport(String resourceCrn, DiagnosticParameters parameters, UsageProto.CDPVMDiagnosticsFailureType.Value failureType,
            Exception exception) {
        if (parameters != null) {
            sendCdpDiagEvent(resourceCrn, parameters, failureType, exception);
        } else {
            LOGGER.debug("Skip sending diagnostics report as diagnostic parameter input is empty.");
        }
    }

    private void sendCdpDiagEvent(String resourceCrn, DiagnosticParameters parameters, UsageProto.CDPVMDiagnosticsFailureType.Value failureType, Exception exception) {
        //LOG
        UsageProto.CDPVMDiagnosticsEvent.Builder eventBuilder = UsageProto.CDPVMDiagnosticsEvent.newBuilder();
        if (exception == null) {
            //LOG
            eventBuilder.setResult(UsageProto.CDPVMDiagnosticsResult.Value.SUCCESSFUL);
        } else {
            //LOG
            eventBuilder.setFailureMessage(StringUtils.left(exception.getMessage(), ERROR_MESSAGE_MAX_LENGTH));
            eventBuilder.setResult(UsageProto.CDPVMDiagnosticsResult.Value.FAILED);
        }
        setIfNotNull(eventBuilder::setFailureType, failureType);
        setIfNotNull(eventBuilder::setUuid, parameters.getUuid());
        setIfNotNull(eventBuilder::setDescription, parameters.getDescription());
        setIfNotNull(eventBuilder::setAccountId, parameters.getAccountId());
        setIfNotNull(eventBuilder::setInputParameters, parameters.toMap().toString());
        setIfNotNull(eventBuilder::setCaseNumber, parameters.getIssue());
        setIfNotNull(eventBuilder::setResourceCrn, resourceCrn);
        UsageProto.CDPVMDiagnosticsDestination.Value dest = convertUsageDestination(parameters.getDestination());
        setIfNotNull(eventBuilder::setDestination, dest);
        //LOG
        usageReporter.cdpVmDiagnosticsEvent(eventBuilder.build());
        //LOG
    }
```
- that return is so unnecessary
- please rename methods to start with a verb
- add more logs